projects/storefrontlib/src/lib/global-message/facade/global-message.service.ts

```diff
@@ -8,19 +8,32 @@ import { GlobalMessage, GlobalMessageType } from '../models/message.model';
 
 @Injectable()
 export class GlobalMessageService {
-  readonly messages$: Observable<
-    Map<GlobalMessageType, string[]>
-  > = this.store.pipe(
-    select(fromStore.getGlobalMessagesEntities),
-    filter(data => data !== undefined)
-  );
-
   constructor(private store: Store<fromStore.GlobalMessageState>) {}
 
+  /**
+   * Get all global messages
+   */
+  getAllMessages(): Observable<Map<GlobalMessageType, string[]>> {
```

Comment:
As mentioned in the ticket description, could this be one of the cases where we can shorten the function name to .get() ?